src/os/filestore/FileStore.cc

```diff
@@ -237,18 +237,22 @@ int FileStore::lfn_open(const coll_t& cid,
 			const ghobject_t& oid,
 			bool create,
 			FDRef *outfd,
-                        Index *index)
+			Index *index,
+			bool directio)
 {
   assert(outfd);
   int r = 0;
   bool need_lock = true;
   int flags = O_RDWR;
+  FDCache *cachep = directio ? &fdcache_direct : &fdcache;
 
   if (create)
     flags |= O_CREAT;
   if (g_conf->filestore_odsync_write) {
     flags |= O_DSYNC;
   }
+  if (directio)
+    flags |= O_DIRECT;
```

Comment:
> The suggested use is only for high-performance storage such as SSDs and newer memory technologies that are coming up.

Those devices are (much) slower than the RAM, so caching reads improves the performance.  I think a thread pool with blocking reads will perform better than Linux' AIO (see https://www.nginx.com/blog/thread-pools-boost-performance-9x)